### PR TITLE
Make compatible with OmniAuth.config.mock_auth

### PR DESCRIPTION
**Pull request by maxcal**
Fixed two issues that make the controller and model authentication break tests using OmniAuth mocks.
See https://github.com/intridea/omniauth/wiki/Integration-Testing

### CODE REVIEW

File: lib/generators/domp/templates/model_authentication.rb [hunk 1]
```diff
@@ -10,8 +10,8 @@ class <%= class_name %>Authentication < ActiveRecord::Base
 
 <% end -%>
   def self.create_from_omniauth(params, <%= class_name.downcase %>, provider)
+    params = HashWithIndifferentAccess.new(params)
```
Comment (AlexanderZaytsev) [inline lib/generators/domp/templates/model_authentication.rb:13]: This is needed because when using mocks, the keys will be symbols, not strings?

File: lib/generators/domp/templates/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -11,6 +11,10 @@ def <%= provider.underscore %>
 
     def create
       auth_params = request.env["omniauth.auth"]
+      unless auth_params.is_a?(OmniAuth::AuthHash)
```
Comment (AlexanderZaytsev) [inline lib/generators/domp/templates/omniauth_callbacks_controller.rb:14]: Why is this needed?

Comment (maxcal) [inline lib/generators/domp/templates/omniauth_callbacks_controller.rb:14]: Because if you use a standard hash to mock the auth response the method will fail when it tries to access the properties with . notation. Either that or just use square brackets when accessing properties from `auth_params`

File: lib/generators/domp/templates/model_authentication.rb (hunk 1 above)
Comment (maxcal) [inline lib/generators/domp/templates/model_authentication.rb:13]: Yes, it allows you to use both.

File: lib/generators/domp/templates/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (AlexanderZaytsev) [inline lib/generators/domp/templates/omniauth_callbacks_controller.rb:14]: But why would you use a standard hash? If you follow recommendations from omniauth's guide and do this, no changes will be neccessary at all (`[:provider]`, `['provider']` and `.provider` will all work):

``` ruby
OmniAuth.config.mock_auth[:twitter] = OmniAuth::AuthHash.new({
    :provider => 'twitter',
    :uid => '123545'
  })
```

**Comment by AlexanderZaytsev** [comment]
Closing this because if you mock properly, the existing code works.

Properly is this:

``` ruby
OmniAuth.config.mock_auth[:twitter] = OmniAuth::AuthHash.new({
    :provider => 'twitter',
    :uid => '123545'
  })
```
